analyzer/src/main/kotlin/managers/Pip.kt

```diff
@@ -318,27 +339,41 @@ class Pip(
 
         val workingDir = definitionFile.parentFile
 
-        // List all packages installed locally in the virtualenv.
-        val pipdeptree = runInVirtualEnv(virtualEnvDir, workingDir, "pipdeptree", "-l", "--json-tree")
+        val jsonFile = createOrtTempDir().resolve("python-inspector.json")
+
+        val pythonInspector = if (definitionFile.name == "setup.py") {
+            PythonInspector.runPythonInspector(
+                workingDir = workingDir,
+                outputFile = jsonFile.absolutePath,
+                setupPyFile = definitionFile.absolutePath,
+                requirementsFile = null
+            )
+        } else {
+            PythonInspector.runPythonInspector(
+                workingDir = workingDir,
+                outputFile = jsonFile.absolutePath,
+                requirementsFile = definitionFile.absolutePath,
+                setupPyFile = null
+            )
+        }
 
         // Get the locally available metadata for all installed packages as a fallback.
         val installedPackages = getInstalledPackagesWithLocalMetaData(virtualEnvDir, workingDir).associateBy { it.id }
 
-        if (pipdeptree.isSuccess) {
-            val fullDependencyTree = jsonMapper.readTree(pipdeptree.stdout)
+        if (pythonInspector.isSuccess) {
+            val fullDependencyTree = jsonMapper.readTree(jsonFile)
+            jsonFile.parentFile.safeDeleteRecursively(force = true)
 
             val projectDependencies = if (definitionFile.name == "setup.py") {
-                // The tree contains a root node for the project itself and pipdeptree's dependencies are also at the
-                // root next to it, as siblings.
+                // The tree contains a root node for the project itself.
```

Comment:
@pombredanne This assumption is not correct anymore with the python-inspector. I tested it locally and the python-inspector output is:

```json
[
  {
    "key": "ply",
    "package_name": "ply",
    "installed_version": "3.11",
    "dependencies": []
  },
  {
    "key": "rdflib",
    "package_name": "rdflib",
    "installed_version": "6.2.0",
    "dependencies": [
      {
        "key": "isodate",
        "package_name": "isodate",
        "installed_version": "0.6.1",
        "dependencies": [
          {
            "key": "six",
            "package_name": "six",
            "installed_version": "1.16.0",
            "dependencies": []
          }
        ]
      },
      {
        "key": "pyparsing",
        "package_name": "pyparsing",
        "installed_version": "3.0.9",
        "dependencies": []
      },
      {
        "key": "setuptools",
        "package_name": "setuptools",
        "installed_version": "65.2.0",
        "dependencies": []
      }
    ]
  }
]
```

So it does not contain a root node with the project name. Therefore the function returns `EMPTY_JSON_NODE` and the spdx-tools-python test is failing. The solution is to simplify lines 367-379 to:

```kotlin
            val projectDependencies = fullDependencyTree.filterNot {
                isPhonyDependency(it["package_name"].textValue(), it["installed_version"].textValueOrEmpty())
            }
```

(and to remove the `projectName` parameter of the function because it is unused now)

The test will still fail because "38" is always passed to python-inspector as Python version and therefore different versions of the dependencies are found. I would suggest to ignore this issue for now and just update the expected results file, and I will address this topic in a separate PR as discussed in the meeting today.